test/integration/settings.test.ts

```diff
@@ -0,0 +1,234 @@
+import chai from 'chai'
+import { solidity, deployMockContract } from 'ethereum-waffle'
+import hre from 'hardhat'
+import {
+  createAndGetLoan,
+  createLoan,
+  fundedMarket,
+  FundedMarketReturn,
+} from '../fixtures'
+import { BigNumberish, Signer } from 'ethers'
+import { AssetSettings, Settings } from '../../types/typechain'
+import { ERC20 } from '@chainlink/contracts/ethers/v0.4/ERC20'
+
+chai.should()
+chai.use(solidity)
+
+const { deployments, getNamedSigner, contracts, ethers, fastForward } = hre
+
+const setupTest = deployments.createFixture(async () => {
+  await deployments.fixture('markets')
+
+  const deployer = await getNamedSigner('deployer')
+
+  // Load settings contract
+  const settings = await contracts.get<Settings>('Settings', { from: deployer })
+
+  // Fund market
+  const market = await fundedMarket()
+
+  return {
+    market,
+    settings,
+  }
+})
+
+describe('Settings', async () => {
+  let market: FundedMarketReturn
+  let settings: Settings
+  let borrower: Signer
+  let borrowerAddress: string
+  let deployer: Signer
+  let newPauserAddress: string
+
+  // Setup for global tests
+  beforeEach(async () => {
+    // Execute snapshot and setup for tests
+    ;({ market, settings } = await setupTest())
+    borrower = await getNamedSigner('borrower')
+    borrowerAddress = await borrower.getAddress()
+    deployer = await getNamedSigner('deployer')
+    newPauserAddress = (await ethers.getSigners())[3].address
+  })
+
+  describe('pauseLendingPool', () => {
+    // Pause a lendingPool and try to take out a loan
+    it('should not be able to take out a loan after platform being paused', async function () {
+      // Pauser lendingPool as deployer
+      ;(
+        await settings
+          .connect(deployer)
+          .pauseLendingPool(market.lendingPool.address)
+      ).should
+        .emit(settings, 'LendingPoolPaused')
+        .withArgs(await deployer.getAddress(), market.lendingPool.address)
+      // Try to take out loan as a borrower
+      const fn = () => createAndGetLoan(market, borrower, 2, hre)
+      await fn().should.be.revertedWith('LENDING_POOL_IS_PAUSED')
+    })
+    it('should not be able to pause the lending pool as not a pauser', async function () {
+      // Try to pause the lending pool
+      await settings
+        .connect(borrower)
+        .pauseLendingPool(market.lendingPool.address)
+        .should.be.revertedWith('NOT_PAUSER')
+    })
+  })
+
+  describe('createAssetSetting', () => {
+    let assetSettings: AssetSettings
+    let newAssetAddress: string
+    let lendingTokenAddress: string
+    let currentMaxLoanSetting: BigNumberish
+    let newCtokenAddress: string
+    let newMaxLoan: number
+    let newMaxTVL: number
+    let newMaxDebt: number
+    let updateValue: number
+
+    before(async () => {
+      // Get asset setting contract from settings
+      assetSettings = await contracts.get('AssetSettings', {
+        at: await settings.assetSettings(),
+      })
```

Comment:
should be able to just do `get('AssetSettings')`

Reply:
changed